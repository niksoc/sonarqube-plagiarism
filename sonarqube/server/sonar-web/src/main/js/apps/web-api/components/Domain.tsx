/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { actionsFilter, getActionKey, Query } from '../utils';
import Action from './Action';
import DeprecatedBadge from './DeprecatedBadge';
import InternalBadge from './InternalBadge';

interface Props {
  domain: T.WebApi.Domain;
  query: Query;
}

export default function Domain({ domain, query }: Props) {
  const filteredActions = domain.actions.filter(action => actionsFilter(query, domain, action));

  return (
    <div className="web-api-domain">
      <header className="web-api-domain-header">
        <h2 className="web-api-domain-title">{domain.path}</h2>

        {domain.deprecatedSince && (
          <span className="spacer-left">
            <DeprecatedBadge since={domain.deprecatedSince} />
          </span>
        )}

        {domain.internal && (
          <span className="spacer-left">
            <InternalBadge />
          </span>
        )}
      </header>

      {domain.description && (
        <div
          className="web-api-domain-description markdown"
          // Safe: comes from the backend
          dangerouslySetInnerHTML={{ __html: domain.description }}
        />
      )}

      <div className="web-api-domain-actions">
        {filteredActions.map(action => (
          <Action
            action={action}
            domain={domain}
            key={getActionKey(domain.path, action.key)}
            showDeprecated={query.deprecated}
            showInternal={query.internal}
          />
        ))}
      </div>
    </div>
  );
}
