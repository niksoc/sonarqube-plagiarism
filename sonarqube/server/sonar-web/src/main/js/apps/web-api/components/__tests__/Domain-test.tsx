/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import Domain from '../Domain';

const ACTION = {
  key: 'foo',
  changelog: [],
  description: 'Foo Desc',
  hasResponseExample: false,
  internal: false,
  post: false
};
const DOMAIN = {
  actions: [ACTION],
  path: 'api',
  description: 'API Desc',
  deprecated: false,
  internal: false
};
const DEFAULT_PROPS = {
  domain: DOMAIN,
  query: { search: '', deprecated: false, internal: false }
};
const SHOW_DEPRECATED = { search: '', deprecated: true, internal: false };
const SHOW_INTERNAL = { search: '', deprecated: false, internal: true };
const SEARCH_FOO = { search: 'Foo', deprecated: false, internal: false };

it('should render deprecated actions', () => {
  const action = { ...ACTION, deprecatedSince: '5.0' };
  const domain = { ...DOMAIN, actions: [action] };
  expect(
    shallow(<Domain {...DEFAULT_PROPS} domain={domain} query={SHOW_DEPRECATED} />)
  ).toMatchSnapshot();
});

it('should not render deprecated actions', () => {
  const action = { ...ACTION, deprecatedSince: '5.0' };
  const domain = { ...DOMAIN, actions: [action] };
  expect(
    shallow(<Domain {...DEFAULT_PROPS} domain={domain} query={SHOW_INTERNAL} />)
  ).toMatchSnapshot();
});

it('should render internal actions', () => {
  const action = { ...ACTION, internal: true };
  const domain = { ...DOMAIN, actions: [action] };
  expect(
    shallow(<Domain {...DEFAULT_PROPS} domain={domain} query={SHOW_INTERNAL} />)
  ).toMatchSnapshot();
});

it('should not render internal actions', () => {
  const action = { ...ACTION, internal: true };
  const domain = { ...DOMAIN, actions: [action] };
  expect(shallow(<Domain {...DEFAULT_PROPS} domain={domain} />)).toMatchSnapshot();
});

it('should render only actions matching the query', () => {
  const actions = [ACTION, { ...ACTION, key: 'bar', description: 'Bar desc' }];
  const domain = { ...DOMAIN, actions };
  expect(
    shallow(<Domain {...DEFAULT_PROPS} domain={domain} query={SEARCH_FOO} />)
  ).toMatchSnapshot();
});

it('should also render actions with a description matching the query', () => {
  const actions = [
    ACTION,
    { ...ACTION, key: 'bar', description: 'Bar desc' },
    { ...ACTION, key: 'baz', description: 'foobar' }
  ];
  const domain = { ...DOMAIN, actions };
  expect(
    shallow(<Domain {...DEFAULT_PROPS} domain={domain} query={SEARCH_FOO} />)
  ).toMatchSnapshot();
});
