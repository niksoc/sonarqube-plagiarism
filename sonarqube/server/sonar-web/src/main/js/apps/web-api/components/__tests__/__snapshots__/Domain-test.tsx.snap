// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should also render actions with a description matching the query 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  >
    <Action
      action={
        Object {
          "changelog": Array [],
          "description": "Foo Desc",
          "hasResponseExample": false,
          "internal": false,
          "key": "foo",
          "post": false,
        }
      }
      domain={
        Object {
          "actions": Array [
            Object {
              "changelog": Array [],
              "description": "Foo Desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "foo",
              "post": false,
            },
            Object {
              "changelog": Array [],
              "description": "Bar desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "bar",
              "post": false,
            },
            Object {
              "changelog": Array [],
              "description": "foobar",
              "hasResponseExample": false,
              "internal": false,
              "key": "baz",
              "post": false,
            },
          ],
          "deprecated": false,
          "description": "API Desc",
          "internal": false,
          "path": "api",
        }
      }
      key="api/foo"
      showDeprecated={false}
      showInternal={false}
    />
    <Action
      action={
        Object {
          "changelog": Array [],
          "description": "foobar",
          "hasResponseExample": false,
          "internal": false,
          "key": "baz",
          "post": false,
        }
      }
      domain={
        Object {
          "actions": Array [
            Object {
              "changelog": Array [],
              "description": "Foo Desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "foo",
              "post": false,
            },
            Object {
              "changelog": Array [],
              "description": "Bar desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "bar",
              "post": false,
            },
            Object {
              "changelog": Array [],
              "description": "foobar",
              "hasResponseExample": false,
              "internal": false,
              "key": "baz",
              "post": false,
            },
          ],
          "deprecated": false,
          "description": "API Desc",
          "internal": false,
          "path": "api",
        }
      }
      key="api/baz"
      showDeprecated={false}
      showInternal={false}
    />
  </div>
</div>
`;

exports[`should not render deprecated actions 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  />
</div>
`;

exports[`should not render internal actions 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  />
</div>
`;

exports[`should render deprecated actions 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  >
    <Action
      action={
        Object {
          "changelog": Array [],
          "deprecatedSince": "5.0",
          "description": "Foo Desc",
          "hasResponseExample": false,
          "internal": false,
          "key": "foo",
          "post": false,
        }
      }
      domain={
        Object {
          "actions": Array [
            Object {
              "changelog": Array [],
              "deprecatedSince": "5.0",
              "description": "Foo Desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "foo",
              "post": false,
            },
          ],
          "deprecated": false,
          "description": "API Desc",
          "internal": false,
          "path": "api",
        }
      }
      key="api/foo"
      showDeprecated={true}
      showInternal={false}
    />
  </div>
</div>
`;

exports[`should render internal actions 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  >
    <Action
      action={
        Object {
          "changelog": Array [],
          "description": "Foo Desc",
          "hasResponseExample": false,
          "internal": true,
          "key": "foo",
          "post": false,
        }
      }
      domain={
        Object {
          "actions": Array [
            Object {
              "changelog": Array [],
              "description": "Foo Desc",
              "hasResponseExample": false,
              "internal": true,
              "key": "foo",
              "post": false,
            },
          ],
          "deprecated": false,
          "description": "API Desc",
          "internal": false,
          "path": "api",
        }
      }
      key="api/foo"
      showDeprecated={false}
      showInternal={true}
    />
  </div>
</div>
`;

exports[`should render only actions matching the query 1`] = `
<div
  className="web-api-domain"
>
  <header
    className="web-api-domain-header"
  >
    <h2
      className="web-api-domain-title"
    >
      api
    </h2>
  </header>
  <div
    className="web-api-domain-description markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "API Desc",
      }
    }
  />
  <div
    className="web-api-domain-actions"
  >
    <Action
      action={
        Object {
          "changelog": Array [],
          "description": "Foo Desc",
          "hasResponseExample": false,
          "internal": false,
          "key": "foo",
          "post": false,
        }
      }
      domain={
        Object {
          "actions": Array [
            Object {
              "changelog": Array [],
              "description": "Foo Desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "foo",
              "post": false,
            },
            Object {
              "changelog": Array [],
              "description": "Bar desc",
              "hasResponseExample": false,
              "internal": false,
              "key": "bar",
              "post": false,
            },
          ],
          "deprecated": false,
          "description": "API Desc",
          "internal": false,
          "path": "api",
        }
      }
      key="api/foo"
      showDeprecated={false}
      showInternal={false}
    />
  </div>
</div>
`;
